crates/subspace-farmer/src/single_plot_farm.rs

```diff
@@ -252,5 +266,30 @@ impl SinglePlotFarm {
             }
             Ok(())
         })
+        .map({
+            let plot = self.plot.clone();
+            move |result| {
+                let sync_sector_size = if total_pieces < max_plot_size {
+                    PieceIndexHashNumber::MAX
+                } else {
+                    PieceIndexHashNumber::MAX / total_pieces * max_plot_size
+                };
+                let expected_range = Range {
+                    start: PieceIndexHashNumber::from_big_endian(&plot.public_key())
+                        .wrapping_sub(&(sync_sector_size / 2)),
+                    end: PieceIndexHashNumber::from_big_endian(&plot.public_key())
+                        .wrapping_add(&(sync_sector_size / 2)),
+                };
+                let got_range = plot.get_piece_range().map(|Range { start, end }| Range {
+                    start: start.into(),
+                    end: end.into(),
+                })?;
+                result.map(|()| SyncResult {
+                    expected_range,
+                    got_range,
+                    total_pieces: plot.piece_count(),
+                })
```

Comment:
I can't say I understand why knowing this ranges is useful at all...
And for number of pieces you can just query the plot directly and get the same result too.

Reply:
Hmm, you right. We probably need only `got_range` here.

The sync range is useful for testing here, we can compare with expected range and check whether the synced range is a subset of our expected one.